### ISSUE TITLE
[BUG] Weird behavior of list of tuples

### ISSUE THREAD

**Issue by OrHayat** [BrianPugh/cyclopts#104]
there is difference between when command need a tuple  and when it expect list of tuples
in unexpected way:
when you handle list of tuples unlike regular tuple if u pass the --item flag it assume that the list is empty  instead of checking that you added at-least one. not sure that intended.

moreover if you pass items and the arity of the tuple is not divisdeable by the number of the params you dont get error it just discard the extra tokens which seems like a bug

was checked on v2.3.0 commit of main

```python
import cyclopts
from cyclopts import Parameter
from typing import Annotated

app = cyclopts.App()


@app.command(name="create")
def create_dict(items: Annotated[list[tuple[str,int|str]],Parameter(name="--item")]):
    """
    item:
        key,value of items to store in the dict - can pass more then one pair or pass the flag multiple times
    """
    res=dict(items)
    print(res)


if __name__=="__main__":
    app()
```

```bash
orhayat:~/cyclopts (main) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   create
╭─ Error ──────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Command "create" parameter "--item,--empty-item" requires 2 arguments. Only got 0.                                   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

orhayat:~/cyclopts (main) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   create  --item
{}

orhayat:~/cyclopts (main) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   create  --item a
{}

orhayat:~/cyclopts (main) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   create  --item a b
{'a': 'b'}

orhayat:~/cyclopts (main) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   create  --item a b c
{'a': 'b'}

orhayat:~/cyclopts (main) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   create  --item a b c 2
{'a': 'b', 'c': 2}

orhayat:~/cyclopts (main) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   create  --item a b c 2 d
{'a': 'b', 'c': 2}

```

**Comment by BrianPugh** [comment]
This is fixed in v2.3.1. If you believe it's not fully fixed we can re-open the issue. Thanks!

**Comment by OrHayat** [comment]
this is better now but still not fixed completly

this is small simple argparse  to show the expected output from list and different input outputs

```python3
import argparse

parser=argparse.ArgumentParser()
parser.add_argument("--items",action='append')
parser.add_argument("--stuff",action='store')
if __name__ == "__main__":
    ns=parser.parse_args()
    print(ns)
```


and some example input output runs
```bash
 > ./cli.py --item a --item b --item
usage: cli.py [-h] [--items ITEMS] [--stuff STUFF]
cli.py: error: argument --items: expected one argument

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py --item a --item b --item c
Namespace(items=['a', 'b', 'c'], stuff=None)

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py --item a --item b --item c --stuff=test
Namespace(items=['a', 'b', 'c'], stuff='test')

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py --item a --item b --item=c --stuff=test
Namespace(items=['a', 'b', 'c'], stuff='test')

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py --item a --item b --item= --stuff=test
Namespace(items=['a', 'b', ''], stuff='test')

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py --item a --item b --item= c --stuff=test
usage: cli.py [-h] [--items ITEMS] [--stuff STUFF]
cli.py: error: unrecognized arguments: c
```


and now  small cyclopts command that send list of tuples

```python3

import cyclopts
cli=cyclopts.App()

@cli.command
def cmd(items:list[tuple[str,str]]=None,stuff:str=None):
    print(f"items:{items},stuff:{stuff}")

if __name__ == "__main__":
    cli()
```



version:
```bash
pip3 list   |grep cyclopts
cyclopts                   2.3.1
```

```python3
import cyclopts
cli=cyclopts.App()

@cli.command
def cmd(items:list[tuple[str,str]]=None,stuff:str=None):
    print(f"items:{items},stuff:{stuff}")
if __name__ == "__main__":
    cli()

```


```bash

 > ./cli.py cmd --items a b --stuff g
items:[('a', 'b')],stuff:g

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py cmd --items a  --stuff g
items:[],stuff:g

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py cmd --items= a  --stuff g
items:[('', 'a')],stuff:g

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py cmd --items= a b --stuff g
items:[('', 'a')],stuff:g


```

**Comment by OrHayat** [comment]
other example on list of strings

```

even if u replace the type of list of tuples to list of string it dont behave as expected

```python3
import cyclopts
cli=cyclopts.App()

@cli.command
def cmd(items:list[str]=None,stuff:str=None):
    print(f"items:{items},stuff:{stuff}")
if __name__ == "__main__":
    cli()

```

```bash
./cli.py cmd --items a b
items:['a', 'b'],stuff:None   #ok

orhayat:~ via 🐹 v1.21.4 via 🐍 v3.10.12
 > ./cli.py cmd --items a b --stuff g
items:['a', 'b'],stuff:g #ok


./cli.py cmd --items=a b --stuff g
items:['a', 'b'],stuff:g # ok

./cli.py cmd --items=a # error but should be [a],None....
╭─ Error ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Command "cmd" parameter "--items,--empty-items" requires an argument.                                                                                                 │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯


./cli.py cmd --items= #error  idk what expected (work on argparse but w/e) but anyway look on the next example
╭─ Error ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Command "cmd" parameter "--items,--empty-items" requires an argument.                                                                                                 │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯


./cli.py cmd --items= a 
items:['', 'a'],stuff:None 

```

**Comment by BrianPugh** [comment]
thanks for the thorough report! I'll work on this tonight.

**Comment by BrianPugh** [comment]
@OrHayat please checkout #109, I believe it should resolve all the issues.

IMHO specifying stuff like `--items= a` should either

1. result in `['', 'a']`. or:
2. result in an error saying that you cannot have a space immediately following an `=`, as this behavior is probably not desirable and could cause confusion.

If you agree with (2), I could add that check.

So similarly, `cmd --items= a b --stuff g` interprets 3 tokens for `--items`: `'', 'a', 'b'`, and so it now appropiately raises an exception that the 3 arguments cannot be divisible by two arguments per element.

**Comment by OrHayat** [comment]
the side branch seems to work as expected.
i think 1 is ok behaviour it the same behaviour as argparse does  and also it the same behaviour in typer.

**Comment by BrianPugh** [comment]
this is fixed in v2.3.2. I think we may revisit the `--option=` discussion in the future, but I will close this issue for now (but we can certainly re-open if/when you find more bugs!).
